source/Octopus.Client/Repositories/Async/EventRepository.cs

```diff
@@ -116,6 +117,15 @@ public EventRepository(IOctopusAsyncClient client)
                 toAutoId,
                 documentTypes
             });
+
+            return Client.List<EventResource>(Client.RootDocument.Link("Events"), parameters);
+        }
+
+        public IEventRepository LimitTo(bool includeGlobal, params string[] spaceIds)
+        {
+            var repository = new EventRepository(Client);
```

Comment:
It feels like this should be more like
```
var parameters = this.CreateParameters(spaceIds, includeGlobal);
return new EventRepository(Client, parameters); // this can be a private constructor
```

where `CreateParameters` in a protected method that looks like

```
protected SpaceQueryParameters CreateParameters(bool includeGlobal, parameters string[] spaceIds) 
{
    var newParameters = new SpaceQueryParameters(){IncludeGlobal = includeGlobal, SpaceIds = spaceIds};
    ParameterHelper.ValidateSpaceParameters(SpaceQueryParameters, newParameters);
    return newParameters;
}
```